### PR TITLE
[fix] calc working day on salary slip when outside default holiday list

### PR DESCRIPTION
**Pull request by RobertSchouten**
fix's issue created when default holiday list is changed to outside the date range of the salary slip

### CODE REVIEW

File: erpnext/hr/doctype/salary_slip/salary_slip.py [hunk 1]
```diff
@@ -252,17 +255,45 @@ def get_payment_days(self, joining_date, relieving_date):
...
-				and holiday_date >= %(start_date)s
+				holiday_date >= %(start_date)s
 				and holiday_date <= %(end_date)s''', {
-					"holiday_list": holiday_list,
 					"start_date": start_date,
 					"end_date": end_date
-				})
+				})[0]
+
+		if count == d_count:
```
Comment (nabinhait) [inline erpnext/hr/doctype/salary_slip/salary_slip.py]: What if there are multiple holiday lists but each one has separate dates. In that case, it should take holiday list from Employee.

Comment (nabinhait) [inline erpnext/hr/doctype/salary_slip/salary_slip.py]: What if any of the holiday list does not apply to a specific employee?

Comment (RobertSchouten) [inline erpnext/hr/doctype/salary_slip/salary_slip.py]: have been thinking about this issue and i think the holiday lists need to be grouped up so you can assign a new holiday list while the old one stays active

**Comment by rmehta** [comment]
@RobertSchouten any update on this?

**Comment by RobertSchouten** [comment]
@rmehta to fix the addressed issues the holiday list requires a rewrite which i dont have time for atm

**Pull request closed without merging**